src/internal/alloc.rs

```diff
@@ -54,11 +54,16 @@ impl<F> Allocator<F> {
             );
         }
         let next_id = self.fat[index];
-        debug_assert!(
-            next_id == consts::END_OF_CHAIN
-                || (next_id <= consts::MAX_REGULAR_SECTOR
-                    && (next_id as usize) < self.fat.len())
-        );
+        if next_id != consts::END_OF_CHAIN
+            && (next_id > consts::MAX_REGULAR_SECTOR
+                || next_id as usize >= self.fat.len())
+        {
+            invalid_data!(
+                "next_id is {}, but should be {}",
```

Comment:
I think this error message is not quite right—it's not that the next ID "should" be `consts::END_OF_CHAIN` specifically, but just that it's invalid in some way.

Reply:
Makes sense. It could be invalid in a few scenarios I guess, so I made it more generic in [496549c](https://github.com/mdsteele/rust-cfb/pull/24/commits/496549c3e8ac034e3da4afb69d4e2eadf077db10)